src/components/RequestCard/index.tsx

```diff
@@ -47,37 +50,123 @@ const RequestCardPlaceholder = () => {
 };
 
 interface RequestCardErrorProps {
-  mediaId?: number;
+  requestData?: MediaRequest;
 }
 
-const RequestCardError = ({ mediaId }: RequestCardErrorProps) => {
+const RequestCardError = ({ requestData }: RequestCardErrorProps) => {
   const { hasPermission } = useUser();
   const intl = useIntl();
 
   const deleteRequest = async () => {
-    await axios.delete(`/api/v1/media/${mediaId}`);
-    mutate('/api/v1/request?filter=all&take=10&sort=modified&skip=0');
+    await axios.delete(`/api/v1/media/${requestData?.media.id}`);
+    mutate(recentlyAddedApi);
+    mutate(recentRequestsApi);
```

Comment:
I dont really like having TItleCard import from Discover. Its a strange cross dependency. Better to just write out the known strings here imo.

Reply:
Hm, okay! I'll revert that change :)